src/main/java/com/rapid7/client/dcerpc/io/PacketInput.java

```diff
@@ -146,4 +148,59 @@ public String readStringBufRef(final boolean nullTerminated) throws IOException
 
         return result;
     }
+
+    /**
+     * See <a href="http://technet.microsoft.com/en-us/library/cc753858(WS.10).aspx>Technet Article</a>
+     * <p>
+     * When creating PSOs, the values of these attributes are in I8 format, which stores time in the intervals of -100
+     * nanoseconds. (Schema: attributeSyntax = 2.5.5.16 (I8).) To set these attributes to appropriate values, convert
+     * time values in minutes, hours, or days to time values in the intervals of 100 nanoseconds, and then precede the
+     * resultant values with a negative sign.
+     * <p>
+     * <table>
+     * <col width="25%"/> <col width="75%"/> <thead>
+     * <tr>
+     * <th>Time Unit</th>
+     * <th>Multiplication Factor</th>
+     * </tr>
+     * </thead> <tbody>
+     * <tr>
+     * <td>m minutes</td>
+     * <td>-60*(10^7) = - 600000000</td>
+     * </tr>
+     * <tr>
+     * <td>h hours</td>
+     * <td>-60*60* (10^7) = -36000000000</td>
+     * </tr>
+     * <tr>
+     * <td>d days</td>
+     * <td>-24*60*60*(10^7) = -864000000000</td>
+     * </tr>
+     * </tbody>
+     * </table>
+     * <p>
+     * For example, if you want to set the msDS-MaximumPasswordAge to 10 days, multiply 10 by -864000000000 and apply the
+     * resulting I8 value to the msDS-MaximumPasswordAge attribute (in this example, -8640000000000). If you want to set
+     * msDS-LockoutDuration to 30 minutes, multiply 30 by -600000000 to get the corresponding I8 value (in this example,
+     * -18000000000).
+     * <p>
+     * INTER_NEVER = {@value}, the constant storing the literal value for an infinite duration (never expire).
+     */
+
+    /**
+     * Interval values are stored negative, in 100 ns  values. To get the interval in seconds,
+     * we must multiply by 10000000 (100ns/sec) and get the absolute value.
+     */
+    public int readInterval() throws IOException {
```

Comment:
I think it would make more sense for reading interval to be implemented in the respective object class, e.g. log off info object.

PacketInput shouldn't care.

Reply:
Actually, is this implementating readBigInteger? We can rename this to readBigInteger and move the logic for handling interval to the respective class.